src/sql-parser/src/parser.rs

```diff
@@ -7048,7 +7053,13 @@ impl<'a> Parser<'a> {
 
         self.expect_keywords(&[FOR, CREATE])?;
 
-        if let Statement::CreateSink(statement) = self.parse_create_sink()? {
+        // parsing if sink name is not provided and also when name is given
+        let create_sink_stmt = match self.maybe_parse(|parser| parser.parse_create_sink(false)) {
+            Some(stmt) => stmt,
+            None => self.parse_create_sink(true)?,
+        };
```

Comment:
This was the cleanest way I could parse this completely so that the sink name is optional. (If the name is not provided,  in `CREATE SINK FROM mv ...`, `from` is a valid identifier name and is parsed as such.)

Do flag if there was a better way to do this.

Reply:
Maybe something like:

```Rust
fn parse_create_sink(&mut self, optional_name: bool) -> Result<Statement<Raw>, ParserError> {
   // ...
   let name = self.parse_item_name();
   let name = if optional_name {
       name.ok()
   } else {
       Some(name?)
   };
   // ...
}

// ...

let create_sink_stmt = self.parse_create_sink(true)?;
```

Reply:
The problem with this was `let name = self.parse_item_name();` would parse `FROM` as the item name and it would fail while trying to look for `FROM` as the keyword after this.

Reply:
(It's actually valid to say `CREATE SINK from FROM ....`

Reply:
Ah, that's another reason why I'd lean towards not including this. We're adding look ahead to parsing the `EXPLAIN`, which I prefer to avoid.

Reply:
Yeah, this was the exact reason why I was unsure.

@benesch Would it make sense to punt this for the time being? See if we have any user feedback first and then make the name optional?

Reply:
I hammered on the code here to try to minimize the complexity required. It's now <10 LOC, which feels to me like a reasonable amount of complexity to take on for the ergonomic win.